internal/errors/traced.go

```diff
@@ -35,6 +35,22 @@ type frameTracer struct {
 	pc uintptr
 }
 
+// Capture is responsible for recording the location where this function was
+// called from in the error supplied. This allows errors that are being passed
+// up through a stack to have extra information attached to them at call sites.
+//
+// Captured errors should only be used in scenario's where adding extra context
+// to an error is not necessary or can't be done.
+//
+// [ErrorStack] can be used to gather all of the capture sites about an error.
+func Capture(err error) Traced {
```

Comment:
I like this, and I don't want to bike shed on the name too much because we want to be able to differentiate between `errors.Trace()` and this. Having said that, could we name it `errors.WithTrace`, or `errors.Stack`?

Reply:
@SimonRichardson what about `errors.Mark(err)`?

Reply:
Land as is, we can always rename